File: python/src/exams/exams2/Exam104.py
""" 104-计算 K 置位下标对应元素的和

"""

from typing import List


def sumIndicesWithKSetBits(nums: List[int], k: int) -> int:
    total = 0

    for i in range(len(nums)):
        s = f'{i:b}'  # 格式化为二进制字符串

        # 统计字符串中 1 的个数
        if s.count('1') == k:
            total += nums[i]

    return total

File: kotlin/src/main/kotlin/exams/exams2/Exam104.kt
package exams.exams2

fun sumIndicesWithKSetBits(nums: List<Int>, k: Int): Int {
    var sum = 0

    for (i in nums.indices) {
        var binStr = i.toString(2)
        var c = binStr.count { it == '1' }

        if (c == k)
            sum += nums[i]
    }

    return sum
}
